src/afa/tests-afa.cc

```diff
@@ -11,11 +11,353 @@ using namespace Mata::Parser;
+	aut.add_trans(1, 1, Nodes{Node{0}, Node{1, 2}});
+	aut.add_trans(2, 0, Nodes{Node{2}, Node{0, 1}});
+	
+    aut.add_inv_trans(0, 0, Nodes{Node{0}});
+	aut.add_inv_trans(0, 1, Nodes{Node{1}});
+	aut.add_inv_trans(1, 1, Nodes{Node{0}, Node{1, 2}});
+	aut.add_inv_trans(2, 0, Nodes{Node{2}, Node{0, 1}});
+
+    REQUIRE(antichain_concrete_forward_emptiness_test(aut));
+    REQUIRE(antichain_concrete_backward_emptiness_test(aut));
+
+	aut.finalstates = {0};
+    REQUIRE(!antichain_concrete_forward_emptiness_test(aut));
+    REQUIRE(!antichain_concrete_backward_emptiness_test(aut));
+
+	aut.finalstates = {1};
+    
+    std::cout << aut.get_initial_nodes();
+    std::cout << aut.post(aut.get_initial_nodes());
+
+    REQUIRE(!antichain_concrete_forward_emptiness_test(aut));
+    REQUIRE(!antichain_concrete_backward_emptiness_test(aut));
+
+    /////////////////////////////////
+    // Example of an automaton
+    /////////////////////////////////
+
+    Afa aut1(10);
+
+	aut1.initialstates = {0};
+	aut1.finalstates = {9};
+
+    aut1.add_trans(0, 0, Nodes{Node{1}});
+    aut1.add_trans(1, 0, Nodes{Node{2}});
+    aut1.add_trans(2, 0, Nodes{Node{3}});
+    aut1.add_trans(3, 0, Nodes{Node{4}});
+    aut1.add_trans(4, 0, Nodes{Node{5}});
+    aut1.add_trans(5, 0, Nodes{Node{6}});
+    aut1.add_trans(6, 0, Nodes{Node{7}});
+    aut1.add_trans(7, 0, Nodes{Node{8}});
+    aut1.add_trans(8, 0, Nodes{Node{8, 9}});
+
+    aut1.add_inv_trans(0, 0, Nodes{Node{1}});
+    aut1.add_inv_trans(1, 0, Nodes{Node{2}});
+    aut1.add_inv_trans(2, 0, Nodes{Node{3}});
+    aut1.add_inv_trans(3, 0, Nodes{Node{4}});
+    aut1.add_inv_trans(4, 0, Nodes{Node{5}});
+    aut1.add_inv_trans(5, 0, Nodes{Node{6}});
+    aut1.add_inv_trans(6, 0, Nodes{Node{7}});
+    aut1.add_inv_trans(7, 0, Nodes{Node{8}});
+    aut1.add_inv_trans(8, 0, Nodes{Node{8, 9}});
+
+    REQUIRE(antichain_concrete_forward_emptiness_test(aut1));
+    REQUIRE(antichain_concrete_backward_emptiness_test(aut1));
+
+    aut1.add_inv_trans(8, 0, Nodes{Node{9}});
+    aut1.add_inv_trans(8, 0, Nodes{Node{9}});
+
+    REQUIRE(!antichain_concrete_forward_emptiness_test(aut));
+    REQUIRE(!antichain_concrete_backward_emptiness_test(aut));
+
+    /////////////////////////////////
+    // Automaton with no transitions
+    /////////////////////////////////
+
+    Afa aut2(3);
+
+    REQUIRE(antichain_concrete_forward_emptiness_test(aut2));
+    REQUIRE(antichain_concrete_backward_emptiness_test(aut2));
+
+    aut2.initialstates = {0};
+
+    REQUIRE(antichain_concrete_forward_emptiness_test(aut2));
+    REQUIRE(antichain_concrete_backward_emptiness_test(aut2));
+
+    aut2.finalstates = {1};
+
+    REQUIRE(antichain_concrete_forward_emptiness_test(aut2));
+    REQUIRE(antichain_concrete_backward_emptiness_test(aut2));
+
+    aut2.finalstates = {0};
+
+    REQUIRE(!antichain_concrete_forward_emptiness_test(aut2));
+    REQUIRE(!antichain_concrete_backward_emptiness_test(aut2));
+
+
+}
+
+
+/*
```

Comment:
Either remove this test, or fix it, but do not keep this test commented out just like this.

And if you decide to remove the test, @martinhruska should know it and be aware that the parsing of AFAs no longer works. Do I understand it right?

Reply:
Thanks for your suggestion, I asked @martinhruska and he said it is ok to leave it commented for now while he is working on the parsing of AFAs.